File: 7 Rings.c
#include<stdio.h>
#include<math.h>
int main()
{
    int t;
    scanf("%d",&t);
    while(t--)
    {
        int n,x,a,d;
        scanf("%d%d",&n,&x);
        a=n*x;
        d=log10(a)+1;
        if(d==5&&d%10!=0)
        printf("YES\n");
        else
        printf("NO\n");
    }
}

File: 7 Rings.java
import java.util.*;
class Codechef
{
    public static void main(String args[])
    {
        Scanner sc=new Scanner(System.in);
        int t=sc.nextInt();
        while(t-- >0)
        {
            int n=sc.nextInt();
            int x=sc.nextInt();
            int a=n*x;
            int d=(int)Math.log10(a)+1;
            if(d==5&&d%10!=0)
            System.out.println("YES");
            else
            System.out.println("NO");
        }
    }
}